examples/XantoI2C_KT0803L_Registers/XantoI2C_KT0803L_Registers.ino

```diff
@@ -53,22 +53,12 @@ uint8_t readRegister(uint8_t register_address) {
   i2c.writeByte(KT0803_CMD_READ);
   XANTO_DEBUG_PRINT("write3 ");
 
-  if (i2c.readAck()) {
-    XANTO_DEBUG_PRINT("ack3 failed ");
-    return 0;
-  } else {
-    XANTO_DEBUG_PRINT("ack3 ");
-  }
+  i2c.writeAck();
 
   uint8_t register_value = i2c.readByte();
   XANTO_DEBUG_PRINT("read ");
 
-  if (i2c.readNack()) {
```

Comment:
I really don't remember why it was working this way. Maybe chip should sends NACK after the last byte to be sent. Need to look into the KT0803 datasheet.

Reply:
on the other hand, here it is indeed a reading from the master, it is therefore up to the master to send the NACK, the code that I have modified should be kept. (this also explains why your code worked because with a bad NACK here at the end of the frame, the transaction stopped but you had already read the data, only the slave understood that it had been badly read, but that should not have great implications).